custom_metrics/prefers-color-scheme.js

```diff
@@ -0,0 +1,53 @@
+// [prefersColorScheme]
+
+const PREFERS_COLOR_SCHEME_REGEXP =
+  /@media\s+\(\s*prefers-color-scheme\s*:\s*(?:dark|light)\s*\)\s*\{[^\}]*\}/gms;
+
+// Checks in three passes:
+// 1. The CSS rules of same-origin and inlined stylesheets in `try {}`.
```

Comment:
Could this be simplified to only use the WPT_BODIES approach? The vast majority of pages in the corpus include some kind of cross-origin stylesheet (80%) so this metric would fall back to the `catch` block most of the time. For the 20% of pages that have no cross-origin stylesheets, is there any advantage to the `document` approach? If it's all the same it'd be good to simplify the code.

```sql
SELECT
  COUNT(DISTINCT IF(type = 'css' AND NET.HOST(url) != NET.HOST(page), pageid, NULL)) / COUNT(DISTINCT pageid) AS pct
FROM
  `httparchive.summary_requests.2021_07_01_mobile`
JOIN
  (SELECT url AS page, pageid FROM `httparchive.summary_pages.2021_07_01_mobile`)
USING (pageid)
```

_(BigQuery doesn't have a `NET.ORIGIN` function so cross-host is used here as a lower bound, but I expect it'll be very close)_

Reply:
I was under the impression the `document` was faster, but given the number of cross-origin stylesheets (thanks for sharing this) have now moved everything to use `$WPT_BODIES` instead (plus a second check to see if `link[rel="stylesheet"][media*="(prefers-color-scheme)"]` matches.